lib/tessel/wifi.js

```diff
@@ -98,8 +98,43 @@ Tessel.prototype.connectToNetwork = function(opts) {
       })
       .then(function() {
         // Then set the new credientials
-        logs.info('Credentials set!');
-        resolve();
+        return self.connection.exec(commands.commitWirelessCredentials());
+      })
+      .then(function(remoteProcess) {
+        // Once the credentials have been comitted
+        remoteProcess.once('close', function() {
+          // Restart the wifi
+          return self.connection.exec(commands.reconnectWifi())
```

Comment:
This is the source of the second test failure—`reconnectWifi` has already been called by `setWifiState` (just after `commitWirelessCredentials`). If you replace `setWifiState` with `self.simpleExec(commands.turnOnWifi(enable))`, then I believe this will work correctly.